plugins/inputs/mqtt_consumer/mqtt_consumer.go

```diff
@@ -313,20 +320,32 @@ func (m *MQTTConsumer) onMessage(acc telegraf.TrackingAccumulator, msg mqtt.Mess
 		}
 		for _, p := range m.TopicParsing {
 			values := strings.Split(msg.Topic(), "/")
+			if len(values) < len(p.SplitTopic) {
+				return fmt.Errorf("config error topic parsing: found topic is shorter than parse topic length")
+			}
```

Comment:
Are you sure this is an error?
Think about a case where you have topic tree with e.g. `/location/device/value` at the one branch but another subtree `/location/device/status/{battery, network, ...}`  on the other branch. If you now apply topic parsing to the later (e.g. to `/location/device/status/+`), the `p.SplitTopic` will always be shorter for the `/location/device/value` topic. Am I missing something?

Reply:
I think it is an error as it is checking that the topic it found is shorter than the topic given. Say the config has the topic `telegraf/testing/#` but the topic that is is found is just `telegraf/testing`. What is found is shorter than what is given so it should error. I added a test for this case. 

Maybe I am misunderstanding your example. Thoughts?

Reply:
Hmm but in your example, what if there is another topic `telegraf/status`? It's also shorter and maybe you just don't want to parse it? Am I missing some previous check or something?

Reply:
You have to explicitly state which topic you want parsed. So another topic that is shorter should not even get looked at.